Fluentify.Web/Controllers/UserController.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Fluentify.Database.Interfaces;
+ using Fluentify.Database.Services;
+using  Fluentify.Models.Database.Tables;
+using Fluentify.Models.Frontend;
+
+
+namespace Fluentify.Web.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class UserController : ControllerBase
+    {
+        private readonly IMapper _mapper;
+        private readonly ILogger<UserController> _logger;
+        private readonly UserService _userService;
+
+        public UserController(IMapper mapper, ILogger<UserController> logger, UserService userService)
+        {
+            _mapper = mapper;
+            _logger = logger;
+            _userService = userService;
+        }
+
+        [HttpGet("{id:int}")]
+        [AllowAnonymous]
+        public async Task<ActionResult<Users>> GetById(int id)
+        {
+            try
+            {
+                var user = await _userService.GetById(id);
+                if (user == null)
+                {
+                    return NotFound();
+                }
+
+                return Ok(_mapper.Map<Users>(user));
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error retrieving user by Id");
+                return StatusCode(500);
+            }
+        }
+
+        [HttpGet]
+        [AllowAnonymous]
+        public async Task<IEnumerable<Users>> GetAll()
+        {
+            return (IEnumerable<Users>)await _userService.GetAll().ToListAsync();
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<ActionResult<Users>> Create(CreateUserRequest createUserRequest)
+        {
+            if (await _userService.UserNameExists(createUserRequest.Nickname))
+            {
+                return BadRequest("Username already exists.");
+            }
+
+            if (await _userService.EmailExists(createUserRequest.Email))
+            {
+                return BadRequest("Email already exists.");
```

Comment:
Consider returning `BadRequest` with error messages: In the `Create` method, when a username or email already exists, you are returning `BadRequest` without any specific error message. It would be helpful to return a more descriptive error message, indicating which field (username or email) is causing the conflict.